src/lxc/conf.c

```diff
@@ -3213,8 +3213,15 @@ static int lxc_execute_bind_init(struct lxc_handler *handler)
 	/* If init exists in the container, don't bind mount a static one */
 	p = choose_init(conf->rootfs.mount);
 	if (p) {
-		free(p);
-		return 0;
+		char *old = p;
+
+		p = strdup(old + strlen(conf->rootfs.mount));
+		if (!p)
```

Comment:
But you need to free(old) in this case right?

Reply:
yes, thanks